### PR TITLE
Fixed memory leak of syslog compression #557 

### PR DESCRIPTION
**Pull request by watchpoints**
### What changes were proposed in this pull request?

 As deleted the fork  by mistake,so I commit again.
before pr 

https://github.com/oceanbase/oceanbase/pull/597

Fixed Fixed memory leak of syslog compression #557
Fixed this file: /oceanbase/deps/oblib/src/lib/oblog/ob_log_compressor.cpp

### Checklist

NO ,

question 
https://open.oceanbase.com/answer/detail?id=13700737

- [ ] I've run the tests to see all new and existing tests pass.
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [ ] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

**Comment by watchpoints** [comment]
image
image

File: deps/oblib/src/lib/oblog/ob_log_compressor.h [hunk 1]
```diff
@@ -29,7 +29,7 @@ class ObLogCompressor final : public lib::ThreadPool {
 public:
   ObLogCompressor();
   virtual ~ObLogCompressor();
-  static ObString get_compression_file_name(const ObString &file_name);
+  static ObString get_compression_file_name(const ObString &file_name,char *buf, const ObString::obstr_size_t buf_size);
```
Comment (JiahuaChen) [inline deps/oblib/src/lib/oblog/ob_log_compressor.h:32]: please add a blank space after the first comma

File: deps/oblib/src/lib/oblog/ob_log.cpp [hunk 2]
```diff
@@ -134,7 +134,16 @@ inline void ob_log_unlink(const char *file_cstr)
 {
   unlink(file_cstr);
   ObString file_name(file_cstr);
-  unlink(ObLogCompressor::get_compression_file_name(file_name).ptr());
+  ObString::obstr_size_t  buf_size = file_name.length() + 1 + sizeof(ObString);
```
Comment (JiahuaChen) [inline deps/oblib/src/lib/oblog/ob_log.cpp:137]: why plus sizeof(ObString), not the ".zst" suffix length？

Comment (watchpoints) [reply deps/oblib/src/lib/oblog/ob_log.cpp]: I write 4,but 4 is not clear, 
so the suffix length of  ".zst" is clear.

File: deps/oblib/unittest/lib/oblog/test_ob_log_compressor.cpp [hunk 3]
```diff
@@ -87,6 +89,7 @@ TEST(ObLogCompressor, normal)
   ASSERT_NE(0, access(file_name.ptr(), F_OK));
   ASSERT_EQ(0, access(compression_file_name.ptr(), F_OK));
   unlink(compression_file_name.ptr());
+  free(compression_file_name_buf);
```
Comment (simonjoylet) [inline deps/oblib/unittest/lib/oblog/test_ob_log_compressor.cpp:92]: if you use `ob_malloc`, then you need `ob_free`

File: deps/oblib/src/lib/oblog/ob_log_compressor.h [hunk 4]
```diff
@@ -24,12 +24,13 @@ namespace common {
...
 class ObString;
 class ObMalloc;
+static const ObString DEFAULT_COMPRESSION_FILE_SUFFIX = ".zst";
 
 class ObLogCompressor final : public lib::ThreadPool {
 public:
   ObLogCompressor();
   virtual ~ObLogCompressor();
-  static ObString get_compression_file_name(const ObString &file_name);
+  static ObString get_compression_file_name(const ObString &file_name, char *buf, const ObString::obstr_size_t buf_size);
```
[the remark this answers is not in the document]
Comment (watchpoints) [reply deps/oblib/src/lib/oblog/ob_log_compressor.h:33]: get_compression_file_name only checks whether the capacity of the input buffer is enough.

the caller of  get_compression_file_name
will  allocate memory and manage content of memory, 
the caller  willing  append  the suffix ?

[the remark this answers is not in the document]
Comment (watchpoints) [reply deps/oblib/src/lib/oblog/ob_log_compressor.h:33]: or the caller need copy the file name to  file_name.and the append suffix in get_compression_file_name

**Pull request closed without merging**